rpc/cluster/FailsafeTask.java

```diff
@@ -41,9 +42,11 @@
     private static final int FETCH_REPLICAS_MAX_RETRIES = 10;
     private static final int WAIT_FOR_PRIMARY_REPLICA_SELECTION_MS = 2000;
     private final ClientClusterRPC client;
+    private final String database;
 
-    FailsafeTask(ClientClusterRPC client) {
+    FailsafeTask(ClientClusterRPC client, String database) {
```

Comment:
We were passing `database` in as a parameter to _every_ method, so it was pretty clear it should've been a ctor parameter.